traits/src/aead/arrayref.rs

```diff
@@ -0,0 +1,58 @@
+//! This module contains the trait and related errors for an Authenticated
+//! Encryption with Associated Data (AEAD) scheme that takes array references
+//! as arguments and writes outputs to mutable array references.
+
+/// An Authenticated Encryption with Associated Data (AEAD) scheme. This trait
+/// is low-level and is mostly used for implementing other, more usable APIs.
+pub trait Aead<const KEY_LEN: usize, const TAG_LEN: usize, const NONCE_LEN: usize> {
```

Comment:
What about keygen?

Reply:
I had removed that because in the derand world it's a little redundant; it would only copy bytes from the rand buffer into they key buffer.